src/lib.rs

```diff
@@ -352,6 +352,93 @@ impl Filter {
     }
 }
 
+/// The possible input methods.
+pub enum InputMethod {
+    /// Read the input from the local Git history.
+    GitHistory,
+
+    /// Read the specified input file.
+    LogFile(String),
+
+    /// Read from `stdin`.
+    Stdin,
+}
+
+impl InputMethod {
+    /// Create a new instance from the given command line options.
+    pub fn parse(matches: &getopts::Matches) -> Option<Self> {
+        let mut method = 0;
+
+        if matches.opt_present("git") {
+            method |= 1
```

Comment:
A XOR-like check for only one method being present:  I assign a certain bit to every input method and set it whether the input method was requested.  If just one bit is set, there was just one input method.  Else, there were either no or more than one input methods requested.

Reply:
Hmm, I feel like there should be a more elegant solution, but I do not see one sadly.

Maybe this behavior could be documented, btw.

Reply:
Okay :wrench:

Reply:
:wrench: :heavy_check_mark: